### ISSUE TITLE
Krita 5.1.5 (flatpak) tries to open a new instance when opening multiple files from the system's file manager

### ISSUE THREAD

**Issue by opulpo** [flathub/org.kde.krita#217]
I experienced this issue running Krita 5.1.5, flatpak version, on Linux Mint 20.3 Cinnamon with default file manager (Nemo 5.2.4).

After opening 2 or more files in Krita, trying to open another Krita file from Nemo launches a new instance of Krita.

Since Krita, as of 5.1.5, cannot run multiple instances anymore, after about 5-10 minutes, a new frozen, unsuable Krita window opens, which has to be force quit. The initial instance is not affected.

To open more than 2 files, one can still click and drag files from Nemo to Krita, or use the File menu within Krita.

Awkwardly, this does not happen before until you try to open more than two files, which shows that somehow flatpak Krita can handle opening files from Nemo within an existing instance.

It seems specific to the Flatpak version, since I didn't experience this problem with the PPA or Appimage versions.

**Comment by vulongm** [comment]
Can reproduce the issue here as well, not entirely sure how to fix this yet. Looks like closing the initial instance also  results in the new instance working.

**Comment by opulpo** [comment]
Indeed, my understanding is that closing the initial instance while the second one is trying to launch unlocks the latter, which was stuck trying to load resources (not allowed for more than one instance).

Not sure if it also unlocks the second instance once it has finally drawn the frozen/unsuable main window.

Anyway, the issue is that no second instance should pop up in the first place.

**Comment by BluesM18A1** [comment]
This is still a problem two years later. Getting much of the same behavior running Bazzite (fedora 41 atomic) + KDE plasma desktop. Just what on earth could cause this?
